lib/v2/leetcode/dailyquestion-solution-en.js

```diff
@@ -106,12 +107,89 @@ module.exports = async (ctx) => {
             headers,
         })
     ).data.data.question.solution;
+    if (article.content === null) {
+        article.content = 'Sorry, the solution of this question may be locked.';
+    }
     const converter = new showdown.Converter();
-    const handleText = (s) => {
-        // 处理多语言代码展示问题
-        s = s.replace(/(```)([a-zA-Z0-9-+#]+)\s*?(\[.*?\])?\n/g, '\r\n###$2\r\n$1$2\r\n');
+    // 图片处理
+    const parsePngSlide = async (s) => {
+        const pattern = /!\?!(.+)!\?!/;
+        if (!pattern.test(s)) {
+            return s;
+        }
+        const matched = s.match(new RegExp(pattern, 'g'));
+        const fn = async (m) => {
+            const relaurl = m.match(pattern)[1].split(':')[0];
+            const fullurl = path.resolve('/' + questionUrl + 'solution/', relaurl).slice(1);
+            const pngList = (
+                await got({
+                    url: fullurl,
+                    method: 'get',
+                    headers,
+                })
+            ).data.timeline;
+            return pngList.map((v) => `![pic](${path.resolve(`/problems/${questionTitle}/solution/`, v.image)})`).join('\n');
+        };
+        const strs = await Promise.all(matched.map((v) => fn(v)));
+        for (let i = 0; i < matched.length; i++) {
+            s = s.replace(matched[i], strs[i]);
+        }
         return s;
     };
+    // iframe代码框处理
+    const parseIframe = async (s) => {
+        const pattern = new RegExp('<iframe.*? src=".*?playground/(.*?)/shared".*</iframe>');
+        if (!pattern.test(s)) {
+            return s;
+        }
+        const matched = s.match(new RegExp(pattern, 'g'));
+        const fn = async (m) => {
+            const uuid = m.match(pattern)[1];
+            const code = (
+                await got({
+                    method: 'post',
+                    url,
+                    json: {
+                        operationName: 'fetchPlayground',
+                        query: `query fetchPlayground {
+                        playground(uuid: "${uuid}") {
+                        testcaseInput
+                        name
+                        isUserOwner
+                        isLive
+                        showRunCode
+                        showOpenInPlayground
+                        selectedLangSlug
+                        isShared
+                        __typename
+                        }
+                        allPlaygroundCodes(uuid: "NPkZEDXK") {
```

Comment:
is this `uuid` supposed to be hardcoded?

Reply:
Sorry, I missed it, but it doesn't seem to matter when I debug it.